packages/ui/src/components/molecules/PopUp/NewPopUp.tsx

```diff
@@ -0,0 +1,161 @@
+ *
+ * Copyright (C) 2023 Axelor (<http://axelor.com>).
+ *
+ * This program is free software: you can redistribute it and/or  modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, {useMemo} from 'react';
+import {StyleSheet, View, Modal} from 'react-native';
+import {useThemeColor} from '../../../theme/ThemeContext';
+import {Card, Text, Icon} from '../../atoms';
+import Button, {ButtonProps} from '../Button/Button';
+
+interface ButtonConfig extends ButtonProps {
+  hide?: boolean;
+}
+interface CancelButtonConfig extends ButtonConfig {
+  showInHeader?: boolean;
+}
+
+interface PopUpProps {
+  style?: any;
+  visible: boolean;
+  title?: string;
+  children: any;
+  cancelButtonConfig?: CancelButtonConfig;
+  confirmButtonConfig?: ButtonConfig;
+  translator?: (key: string) => string;
+}
+
+const PopUp = ({
+  style,
+  visible,
+  title,
+  children,
+  cancelButtonConfig,
+  confirmButtonConfig,
+  translator,
+}: PopUpProps) => {
+  const Colors = useThemeColor();
+
+  const _cancelButtonConfig = useMemo(() => {
+    if (!cancelButtonConfig) {
+      return null;
+    }
+
+    return {
+      ...cancelButtonConfig,
+      title: cancelButtonConfig.title ?? translator('Base_Cancel'),
+      color: cancelButtonConfig.color ?? Colors.errorColor,
+      iconName: cancelButtonConfig.iconName ?? 'times',
+      hide: cancelButtonConfig.hide ?? false,
+      showInHeader: cancelButtonConfig.showInHeader ?? false,
+      width: cancelButtonConfig.width ?? 115,
+    };
+  }, [cancelButtonConfig, translator, Colors]);
+
+  const _confirmButtonConfig = useMemo(() => {
+    if (!confirmButtonConfig) {
+      return null;
+    }
+
+    return {
+      ...confirmButtonConfig,
+      title: confirmButtonConfig.title ?? translator('Base_Ok'),
+      color: confirmButtonConfig.color ?? Colors.primaryColor,
+      iconName: confirmButtonConfig.iconName ?? 'check',
+      hide: confirmButtonConfig.hide ?? false,
+      width: confirmButtonConfig.width ?? 115,
+    };
+  }, [confirmButtonConfig, translator, Colors]);
+
+  return (
+    <Modal
+      visible={visible}
+      transparent
+      animationType="fade"
+      onRequestClose={_cancelButtonConfig?.onPress}>
+      <View style={styles.modalBackground}>
+        <Card style={[styles.container, style]}>
+          <View style={styles.headerContainer}>
+            {!!title && (
+              <Text writingType="title" fontSize={20}>
+                {translator ? translator(title) : title}
```

Comment:
I'm not sure translator is needed here, title will already be translated
If you want a translation key instead of a title, please rename in titleKey but I'm not sure this is needed here